source/common/upstream/maglev_lb.cc

```diff
@@ -0,0 +1,45 @@
+#include "common/upstream/maglev_lb.h"
+
+namespace Envoy {
+namespace Upstream {
+
+MaglevTable::MaglevTable(const HostVector& hosts) {
+  // Implementation of pseudocode listing 1 in the paper (see header file for more info).
+  std::vector<TableBuildEntry> table_build_entries;
+  table_build_entries.reserve(hosts.size());
+  for (const auto& host : hosts) {
+    const std::string& address = host->address()->asString();
+    table_build_entries.emplace_back(HashUtil::xxHash64(address) % TableSize,
```

Comment:
This doesn't need to be part of this change, but should we consider putting a hash function directly on the address, instead of converting to a string?  That would avoid a ton of allocations in this loop.

Reply:
It uses a string view, so there is no copying at all. (The asString() function actually returns a reference to an already allocated string).

Reply:
Oh right, I forgot that was pre-computed. Seems fine as-is then.